plugin/ochttp/trace.go

```diff
@@ -149,11 +150,21 @@ func spanNameFromURL(req *http.Request) string {
 }
 
 func requestAttrs(r *http.Request) []trace.Attribute {
-	return []trace.Attribute{
-		trace.StringAttribute(PathAttribute, r.URL.Path),
-		trace.StringAttribute(HostAttribute, r.URL.Host),
-		trace.StringAttribute(MethodAttribute, r.Method),
-		trace.StringAttribute(UserAgentAttribute, r.UserAgent()),
+	userAgent := r.UserAgent()
+
+	if userAgent != "" {
+		return []trace.Attribute{
+			trace.StringAttribute(PathAttribute, r.URL.Path),
```

Comment:
Why are we doing this?

Reply:
just to omit empty `http.user_agent` attribute. If we have a spec we need to agree on single implementation. Python never collects it, Go - always. In C# I want to omit including empty-values attribute to save space. I'll add a note to spec in specs PR.

Do you disagree with intent or just curious why?

Reply:
I don't disagree with the intent but I prefer not to change things unless there is a reason: it's always possible someone was relying on the existing behavior in a way we didn't anticipate and we create some problems for them.

An additional concern was the code duplication that was introduced here, which we should avoid regardless of the outcome.

I am happy to make this change if it brings Go into better alignment with the other implementations, since it is pretty small. In that case please refactor to remove the duplicate code if possible.

Reply:
In `Go` - will it be better to return a slice of fixed length array to implement this kind of variability?

Reply:
Right now as I said it's not aligned at all. So this is an attempt to do it. I'd like to make this change instead of introducing empty `user_agent` property in other languages. I figured impact here should be small.

I definitely share the desire to minimize change

Reply:
You can do something like:
```go
attrs := make([]trace.Attribute, 0, 4)
attrs = append(attrs, 
  trace.StringAttribute(HostAttribute, r.URL.Host),
  trace.StringAttribute(MethodAttribute, r.Method),
  ...)

if ua != "" {
attrs = append(attrs, trace.StringAttribute(UserAgentAttribute, ua))
}
```
I think that will avoid an extra allocation in the common case where there is a user agent.